client/ayon_maya/plugins/publish/increment_current_file_deadline.py

```diff
@@ -16,25 +16,25 @@ class IncrementCurrentFileDeadline(plugin.MayaContextPlugin):
     targets = ["local"]
 
     def process(self, context):
-
-        from ayon_core.lib import version_up
-        from ayon_core.pipeline.publish import get_errored_plugins_from_context
-        from maya import cmds
-
-        errored_plugins = get_errored_plugins_from_context(context)
-        if any(plugin.__name__ == "MayaSubmitDeadline"
-                for plugin in errored_plugins):
-            raise RuntimeError("Skipping incrementing current file because "
-                               "submission to deadline failed.")
-
         current_filepath = context.data["currentFile"]
         new_filepath = version_up(current_filepath)
 
-        # # Ensure the suffix is .ma because we're saving to `mayaAscii` type
-        if new_filepath.endswith(".ma"):
-            fileType = "mayaAscii"
-        elif new_filepath.endswith(".mb"):
-            fileType = "mayaBinary"
-
-        cmds.file(rename=new_filepath)
-        cmds.file(save=True, force=True, type=fileType)
+        host: IWorkfileHost = registered_host()
+        if hasattr(host, "save_workfile_with_context"):
+            from ayon_core.host.interfaces import SaveWorkfileOptionalData
+            host.save_workfile_with_context(
+                filepath=new_filepath,
+                folder_entity=context.data["folderEntity"],
+                task_entity=context.data["taskEntity"],
+                description="Incremented by publishing.",
```

Comment:
Do we want this description set? And maybe it should be more descriptive if we do? e.g. include from which version it was incremented perhaps, or?